profile/src/main/java/com/scottlogic/datahelix/generator/profile/validators/profile/constraints/grammatical/ConditionalConstraintValidator.java

```diff
@@ -33,7 +34,9 @@ public ConditionalConstraintValidator(List<FieldDTO> fields)
     public ValidationResult validate(ConditionalConstraintDTO conditionalConstraint)
     {
         ValidationResult validateIfConstraint = validateConstraint(conditionalConstraint.ifConstraint, fields);
-        ValidationResult validateThenConstraint = validateConstraint(conditionalConstraint.thenConstraint, fields);
+        ValidationResult validateThenConstraint = conditionalConstraint.thenConstraint == null
```

Comment:
Should this result be thrown here, or within `validateConstraint()`?

Reply:
I had considered this.

My reasoning was that this is logic is specific to the 'ConditionalConstraint' class and encapsulating it there prevents this additional logic from leaking into the more general 'ConstraintValidator' class.